test/compress/template-string.js

```diff
@@ -0,0 +1,162 @@
+template_strings: {
+    beautify = {
+        quote_style: 3
+    }
+    input: {
+        ``;
+        `xx\`x`;
+        `${ foo + 2 }`;
+        ` foo ${ bar + `baz ${ qux }` }`;
+    }
+    expect_exact: "``;`xx\\`x`;`${foo+2}`;` foo ${bar+`baz ${qux}`}`;";
+}
+
+template_string_prefixes: {
+    beautify = {
+        quote_style: 3
+    }
+    input: {
+        String.raw`foo`;
+        foo `bar`;
+    }
+    expect_exact: "String.raw`foo`;foo`bar`;";
+}
+
+template_strings_ascii_only: {
+    beautify = {
+        ascii_only: true,
+        quote_style: 3
+    }
+    input: {
+        var foo = `foo
+        bar
+        ↂωↂ`;
+        var bar = `\``;
+    }
+    expect_exact: "var foo=`foo\\n        bar\\n        \\u2182\\u03c9\\u2182`;var bar=`\\``;"
+}
+
+template_strings_without_ascii_only: {
+    beautify = {
+        quote_style: 3
+    }
+    input: {
+        var foo = `foo
+        bar
+        ↂωↂ`
+    }
+    expect_exact: "var foo=`foo\n        bar\n        ↂωↂ`;"
+}
+
+template_string_with_constant_expression: {
+    options = {
+        evaluate: true
+    }
+    beautify = {
+        quote_style: 3
+    }
+    input: {
+        var foo = `${4 + 4} equals 4 + 4`;
+    }
+    expect: {
+        var foo = `8 equals 4 + 4`;
+    }
+}
+
+template_string_with_predefined_constants: {
+    options = {
+        evaluate: true
+    }
+    beautify = {
+        quote_style: 3
+    }
+    input: {
+        var foo = `This is ${undefined}`;
+        var bar = `This is ${NaN}`;
+        var baz = `This is ${null}`;
+        var foofoo = `This is ${Infinity}`;
+        var foobar = "This is ${1/0}";
+        var foobaz = 'This is ${1/0}';
+        var bazbaz = `This is ${1/0}`;
+    }
+    expect: {
+        var foo = `This is undefined`;
+        var bar = `This is NaN`;
+        var baz = `This is null`;
+        var foofoo = `This is Infinity`;
+        var foobar = "This is ${1/0}";
+        var foobaz = 'This is ${1/0}';
+        var bazbaz = `This is Infinity`;
```

Comment:
This one should have remained:
```
        var bazbaz = `This is ${1/0}`;
```
because it is shorter.

Add a test for:
```
var z = `This is ${3/7}`
```